client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/HotrodCache.java

```diff
@@ -127,4 +145,80 @@ protected boolean isStarted() {
         });
     }
 
+    @Override
+    protected <T> Future<T> withCache(
+            final Function<org.infinispan.commons.api.BasicCache<K, V>, CompletionStage<T>> futureSupplier) {
+        return super.withCache(futureSupplier)
```

Comment:
IMHO invoking the super class implementation is an anti pattern as you have no control over what is happening there.
FMPOV *withCache* should be declared final and should be changed to invoke e.g. a *postCacheAccess(Handler<AsyncResult<?>>)* method with the outcome of the cache access. Subclasses can then override this method or not. But its purpose is well defined in any case ...

Reply:
Yes, that's better. I've changed it accordingly.